src/rgw/rgw_rest_usage.cc

```diff
@@ -5,13 +5,137 @@
 #include "rgw_usage.h"
 #include "rgw_rest_usage.h"
 #include "rgw_sal.h"
-
+#include "rgw_sal_rados.h"
 #include "include/str_list.h"
 
 #define dout_subsys ceph_subsys_rgw
 
 using namespace std;
 
+
+static inline RGWObjCategory   rgw_obj_get_category_by_name(string name)
+{
+	std::string::iterator end_pos = std::remove(name.begin(), name.end(), ' ');
+	name.erase(end_pos, name.end()); 
+	 
+     if(name.compare("rgw.none") == 0)
+			return  RGWObjCategory::None;
+     if(name.compare("rgw.main") == 0)
+			return RGWObjCategory:: Main;
+     if(name.compare("rgw.shadow") == 0)
+			return RGWObjCategory:: Shadow;
+     if(name.compare("rgw.multimeta") == 0)
+			return RGWObjCategory:: MultiMeta;
+     if(name.compare("InfrequentAccess") == 0)
+			return RGWObjCategory:: InfrequentAccess;
+	 if(name.compare(RGW_STORAGE_CLASS_STANDARD) == 0)
+	 		return RGWObjCategory:: Main;
+	return RGWObjCategory::None;
+}
+
+
+class RGWOp_StorageClass_Get : public RGWRESTOp {
+
+public:
+
+	uint64_t ScSize;
+	RGWOp_StorageClass_Get() {}
+
+	int check_caps(const RGWUserCaps& caps) override {
+		return caps.check_cap("usage", RGW_CAP_READ);
+	}
+	void execute(optional_yield y) override;
+	const char* name() const override { return "get_storageclass"; }
+	void send_response() override;
+};
+
+
+void RGWOp_StorageClass_Get::execute(optional_yield y) { 
+	string sc = s->info.args.get("HTTP_X_AMZ_STORAGE_CLASS");
+	
+	string bucket_name ;
+	string  tenant_name ; 
+	RGWBucketInfo bucket_info;
+	map<RGWObjCategory, RGWStorageStats> stats;   
+	map<string, bufferlist> attrs;
+	
+	
+	//RESTArgs::get_string(s, "uid", uid_str, &uid_str);
+	RESTArgs::get_string(s, "bucket", bucket_name, &bucket_name);
+
+	real_time mtime;
+	rgw::sal::RadosStore* st = static_cast<rgw::sal::RadosStore*>(store);
```

Comment:
This is all fairly easily doable via SAL.  I don't think we're going to merge new code that is RADOS only at this point.